view/src/components/deal.js

```diff
@@ -0,0 +1,479 @@
+    padding: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
+  },
+  search: {
+    position: "relative",
+    borderRadius: theme.shape.borderRadius,
+    marginLeft: "0px",
+    width: "100%",
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: "100%",
+    position: "absolute",
+    pointerEvents: "none",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  inputRoot: {
+    color: "inherit",
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    width: "100%",
+  },
+});
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+/**
+ * This class represents a Deal component, which is a sub-page of the
+ * home page where deal objects are visualized, created, updated, edited,
+ * and deleted.
+ */
+class Deal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // Deal states (haven't been connected to CRUD)
+      deals: "",
+      title: "",
+      body: "",
+      todoId: "",
+      // Page states
+      errors: [],
+      open: false,
+      uiLoading: true,
+      buttonType: "",
+      viewOpen: false,
+    };
+
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleEditClick = this.handleEditClick.bind(this);
+    this.handleViewOpen = this.handleViewOpen.bind(this);
+  }
+
+  /**
+   * Given an event, this function updates a state (the target of the event)
+   * with a new value
+   * @param event The event that is attempting to update a state
+   */
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  /** Returns the authentication token stored in local storage */
+  getAuth = () => {
+    authMiddleWare(this.props.history);
+    return localStorage.getItem("AuthToken");
+  };
+
+  /** Load in all of the current deals when the component has mounted */
+  componentDidMount() {
+    axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
+    axios
+      .get("/todos")
+      .then((response) => {
+        this.setState({
+          // Produce state
+          deals: response.data,
+          // Page state
+          uiLoading: false,
+        });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  /**
+   * Takes a deal object as an input and deletes the given deal
+   * object from the database
+   * @param data A deal object
+   */
+  handleDelete(data) {
+    authMiddleWare(this.props.history);
+    const authToken = localStorage.getItem("AuthToken");
+    axios.defaults.headers.common = { Authorization: `${authToken}` };
+    let todoId = data.todo.todoId;
+    axios
+      .delete(`todo/${todoId}`) // NOT CONNECTED TO DEALS CRUD YET
```

Comment:
```suggestion
      .delete(`todo/${todoId}`)  // NOT CONNECTED TO DEALS CRUD YET
```
Instead of this comment, maybe something like `TODO(your github username): Connect to Deals backend.`